uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -77,288 +50,131 @@ define([
-     */
-    vm.isPublicHoliday = function (date) {
-      return !!publicHolidays[getDateObjectWithFormat(date).valueOf()];
-    };
+        month = _.find(contact.calendarData, function (month) {
+          return (month.month === monthObj.month) && (month.year === monthObj.year);
+        });
 
-    /**
-     * Labels the given period according to whether it's current or not
-     *
-     * @param  {AbsencePeriodInstance} period
-     * @return {string}
-     */
-    vm.labelPeriod = function (period) {
-      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+        return month ? month.data : [];
+      }
     };
 
     /**
      * Refresh contacts and calendar data
      */
     vm.refresh = function () {
       vm.loading.calendar = true;
-      loadContacts()
+      vm._loadContacts()
         .then(function () {
-          loadLeaveRequestsAndCalender()
+          vm._loadLeaveRequestsAndCalender()
             .then(function () {
               vm.loading.calendar = false;
             });
         });
     };
 
-    (function init() {
-      vm.loading.page = true;
-      //Select current month as default
-      vm.selectedMonths = [vm.months[moment().month()].label];
-      $q.all([
-        loadAbsencePeriods(),
-        loadAbsenceTypes(),
-        loadPublicHolidays(),
-        loadRegions(),
-        loadDepartments(),
-        loadLocations(),
-        loadLevelTypes(),
-        loadStatuses(),
-        loadDayTypes()
-      ])
-      .then(function () {
-        return loadManagees();
-      })
-      .then(function () {
-        vm.legendCollapsed = false;
-        return loadLeaveRequestsAndCalender();
-      })
-      .finally(function () {
-        vm.loading.page = false;
-      });
-
-      $rootScope.$on('LeaveRequest::updatedByManager', function () {
-        vm.refresh();
-      });
-    })();
-
-    /**
-     * Converts given date to moment object with server format
-     *
-     * @param {Date/String} date from server
-     * @return {Date} Moment date
-     */
-    function getDateObjectWithFormat(date) {
-      return moment(date, sharedSettings.serverDateFormat).clone();
-    }
-
     /**
-     * Returns the styles for a specific leaveRequest
-     * which will be used in the view for each date
-     *
-     * @param  {object} leaveRequest
-     * @param  {object} dateObj - Date UI object which handles look of a calendar cell
-     * @return {object}
+     * Fetch all the months from the current period and
+     * save it in vm.months
      */
-    function getStyles(leaveRequest, dateObj) {
-      var absenceType,
-        status = leaveRequestStatuses[leaveRequest.status_id];
-
-      if (status.name === 'waiting_approval'
-        || status.name === 'approved'
-        || status.name === 'admin_approved') {
-        absenceType = _.find(vm.absenceTypes, function (absenceType) {
-          return absenceType.id == leaveRequest.type_id;
+    vm._fetchMonthsFromPeriod = function () {
```

Comment:
It seems to me that this is the same method that you have in the other child calendar, why isn't this in the parent?